backend/src/services/smsService.js

```
@@ -106,6 +106,190 @@ class SMSService {
+  }
+
+  /**
+   * Validate required fields in data object
+   */
+  validateRequiredFields(data, requiredFields, dataType = 'data') {
+    if (!data || typeof data !== 'object') {
+      throw new Error(`${dataType} is required and must be an object`);
+    }
+    
+    const missingFields = requiredFields.filter(field => {
+      return !data.hasOwnProperty(field) || data[field] === null || data[field] === undefined || data[field] === '';
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)